dalesbred/src/main/java/org/dalesbred/query/SqlQuery.java

```diff
@@ -112,6 +120,47 @@ private SqlQuery(@NotNull @SQL String sql, @NotNull List<?> args) {
         return args;
     }
 
+
+    /**
+     * Returns the fetch size of this query
+     */
+    public @Nullable Integer getFetchSize() {
+        return fetchSize;
+    }
+
+    /**
+     * A non-null fetch size will be set as the fetch size for the statements executed from this query.
+     *
+     * @param size fetch size in rows or null
+     * @throws IllegalArgumentException if size is < 0
+     */
+    public void setFetchSize(@Nullable Integer size) {
+        if (size != null && size < 0)
+            throw new IllegalArgumentException("Illegal fetch size " + size + ". Fetch size must be null or > 0");
```

Comment:
The message should have _"or >= 0"_ instead of _"or > 0"_.

And looking at this, I think it's a good thing to get rid of `null` values here.

Before looking at the `PreparedStatement.setFetchSize` API I was about to comment that `setFetchSize(0)` makes no sense, but of course it does. It's probably a good idea to document the special meaning of zero here.